### PR TITLE
Fix #14 - Using Hibernate ORM tool

### LINKED ISSUE #14: Use ORM tool
Use an ORM tool(Hibernate) for managing the database connectivity While the initial setup was manageable with a direct calls from the JDBC, the validator has grown in complexity which makes it difficult to maintain. In the next major version, Hibernate should be incorporated in order to make thinks easier to maintain

### PR DESCRIPTION
**Pull request by mohangandhiGH**
Using Hibernate ORM to manage database.

Fix #14

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,6 +18,10 @@
             <id>public.onebusaway.org</id>
             <url>http://nexus.onebusaway.org/content/groups/public/</url>
         </repository>
+        <repository>
```
Comment (barbeau) [inline pom.xml]: What is this repo for?

Comment (mohangandhiGH) [inline pom.xml]: @barbeau that repository was there before this commit and contains artifacts needed for 

        <dependency>
            <groupId>org.onebusaway</groupId>
            <artifactId>onebusaway-gtfs</artifactId>
            <version>1.3.4</version>
        </dependency>

        
        <dependency>
            <groupId>org.onebusaway</groupId>
            <artifactId>onebusaway-gtfs-hibernate</artifactId>
            <version>1.3.4</version>
            <exclusions>
                <exclusion>
                    <groupId>org.apache.commons</groupId>
                    <artifactId>collections</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

        <dependency>
            <groupId>org.onebusaway</groupId>
            <artifactId>onebusaway-gtfs-realtime-exporter</artifactId>
            <version>1.0.1</version>
        </dependency>

Comment (mohangandhiGH) [inline pom.xml]: @barbeau I thought, you think that repo is newly added in this commit :)

Comment (barbeau) [inline pom.xml]: Here's the repo I was referring to:

~~~
<id>scm.umiacs.umd.edu</id>
<url>https://scm.umiacs.umd.edu/maven/adapt/content/repositories/releases/</url>
~~~

Comment (barbeau) [inline pom.xml]: Interesting, I didn't know that SQLite isn't officially supported as a hiberate dialect.  Looks like MySQL and PostGIS are supported:
https://developer.jboss.org/wiki/SupportedDatabases2?_sscc=t

In theory I believe we should be able to easily switch database providers - do you know of any downsides to switching to an officially supported dialect for a provider like MySQL?

Comment (mohangandhiGH) [inline pom.xml]: I don't see any downsides switching to provider like MySQL, rather it has performance improvements.
One link I pulled online http://www.jpab.org/Hibernate/MySQL/server/Hibernate/SQLite/embedded.html

Shall we use mysql instead sqlite? 
If we switch, for now what I can see are, some query changes.

Comment (barbeau) [inline pom.xml]: Ah - but I guess MySQL wouldn't be embedded, then, right?  You'd need to install/setup MySQL database independently of the gtfs-realtime-validator?

I'd prefer to stick to embedded database options to make this tool as easy as possible to deploy.  What do you think the best embedded option would be that also has official Hibernate dialect support?  Maybe HSQLDB?

Nice find on the performance measurement benchmarks, by the way.

Comment (barbeau) [inline pom.xml]: Here's a good summary of them all - http://www.jpab.org/All/All/All.html.

**Comment by mentaer** [comment]
Yes, better to stick with sqlite which is lightweight and deployable in comparison to a full  blown mysql db.

File: pom.xml (hunk 1 above)
Comment (mohangandhiGH) [inline pom.xml]: 👍

**Pull request merged**